### ISSUE TITLE
Typing the date doesn't follow set format

### ISSUE THREAD

**Issue by peenorolampi** [charliekassel/vuejs-datepicker#486]
If I have format set for example to "d.M.yyyy", like this:

`<datepicker format="d.M.yyyy" :typeable="true">`

And then I select 1st of June 2017, it shows up correctly as "1.6.2017", but then if I only change the year by manually typing "1.6.2018", it interprets the changed date as 6th of January 2018.

**Comment by PavelGonzales** [comment]
Same problem

**Comment by charliekassel** [comment]
Yes internally I parse the dates with Date.parse, so this will follow your locale I presume rather than the format you set in the datepicker.

**Comment by jrglaber** [comment]
I have the same problem. I want to write in pt-br format, but because my browser is defaul en, it does not let me use the correct format.

**Comment by brigmanbr** [comment]
I have the same problem.

`<datepicker :language="ptBR" format="dd/MM/yyyy" v-mask="'##/##/####'" :typeable="true" class="form-control" v-model="dt_ini" @keyup.enter="doFilter" ></datepicker>`

If I type 15/06/2018 then press TAB, my input is cleaned.

**Comment by julienmonty** [comment]
Same here, so something like this is not possible as well (select month and year only with typeable) :

`<datepicker v-model="product.release_date" format="MM/yyyy" :minimumView="'month'" :maximumView="'month'" :typeable="true"></datepicker>`

**Comment by RipkensLar** [comment]
I have the same problem. Would be grateful for a bug fix.

**Comment by charliekassel** [comment]
No bug, this is a new feature.
Happy to look at PRs as I don’t have time right now to develop this.

**Comment by RipkensLar** [comment]
@charliekassel I wasn't talking about the problem from @julienmonty . I meant that I have the same problem as @peenorolampi .

**Comment by charliekassel** [comment]
This is down to javascript Date parse @RipkensLar interpreting in your locale and is directly related to us using the prop format to interpret the date.
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/parse

**Comment by brurubio** [comment]
I found a solution to the `Date.parse` problem. I don't know if it's the best approach, but it solved my problem, and I hope it can help someone else. I created a pullRequest #536. @charliekassel
